libs/community/langchain_community/document_loaders/directory.py

```diff
@@ -124,7 +124,14 @@ def lazy_load(self) -> Iterator[Document]:
         if not p.is_dir():
             raise ValueError(f"Expected directory, got file: '{self.path}'")
 
-        paths = p.rglob(self.glob) if self.recursive else p.glob(self.glob)
+        # glob multiple patterns if a list is provided, e.g., multiple file extensions
+        if type(self.glob) == list:
```

Comment:
Maybe this?

```
if isinstance(self.glob, (list, tuple)):
  ...
elif isinstance(self.glob, str):
  ...
else:
  raise TypeError('Expected glob to be str or sequence of str, but got {type(self.glob)})
```

Reply:
Sounds good! I will make that change on my fork soon. Thank you